qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -710,15 +745,29 @@ def abort_measurement(self) -> None:
         """
         self.write('ABORt')
 
-    def _licenses(self):
-        licenses_raw = self.ask('SYST:LIC:CAT?')
-        licenses_list = [x.strip('"') for x in licenses_raw.split(',')]
-        return licenses_list
+    def _licenses(self) -> Sequence[str]:
+        """
+        Return extra licenses purchased with the DMM. The 34410A does not have
+        optional modules, hence always returns an empty tuple.
+        """
+        if self.model != '34410A':
+            licenses_raw = self.ask('SYST:LIC:CAT?')
+            licenses_list = [x.strip('"') for x in licenses_raw.split(',')]
+            return licenses_list
+        return tuple()
+
+    def _get_parameter(self, sense_function: str = "DC Voltage") -> float:
+        """
+        Measure the parameter given by sense_function
 
-    def _get_voltage(self):
-        # TODO: do we need to set any other parameters here?
+        Args:
+            sense_function: The parameter to measure. Valid values are those
+                accepted by the sense_function parameter.
 
-        with self.sense_function.set_to('DC Voltage'):
+        Returns:
+            The float value of the parameter.
+        """
+        with self.sense_function.set_to(sense_function):
```

Comment:
In the long run I think we are planing a more fundamental fix for this issue but I think this is a nice improvement as is. @WilliamHPNielsen  @GateBuilder What do you think. I guess this will slightly slowdown the measurements ?

Reply:
I believe this adds some communication overhead, but as you pointed out, I expect that to be a very slight one. Overall, this PR is a nice improvement.

Reply:
@jenshnielsen @GateBuilder which "this issue" are you referring to? Do you mean setting sense_function each time "single-point" kind of measurement is done with these parameters?

IMHO, this is OK as a convenience, and a convenience MAY come with extra overhead. In case the user wants to avoid the overhead, he can always resort to using more verbose/explicit API (`init_measurement`+`fetch`).

Reply:
Thanks for checking :) FWIW this is based on the previous code for the `volt` parameter, so even with the overhead, it won't be a regression in relation to the previous versions.